### PR TITLE
Adds routing handler unit tests

### PR DESCRIPTION
**Pull request by saurav-c**
This replaces PR (#104).

### CODE REVIEW

File: tests/bedrock/route/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+CMAKE_MINIMUM_REQUIRED(VERSION 3.6)
+
+SET(KVS_SRC_DIR ${CMAKE_SOURCE_DIR}/src/bedrock/route)
```
Comment (vsreekanti) [inline tests/bedrock/route/CMakeLists.txt]: Nit: This variable should be `ROUTE_SRC_DIR`.

File: tests/bedrock/route/routing_handler_base.hpp [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  zmq::context_t context;
+  SocketCache pushers = SocketCache(&context, ZMQ_PUSH);
+
+  RoutingHandlerTest() { global_hash_ring_map[1].insert(ip, thread_id); }
+
+ public:
+  void SetUp() {
+    // reset all global variables
+    kDefaultLocalReplication = 1;
+    kSelfTierId = 1;
```
Comment (vsreekanti) [inline tests/bedrock/route/routing_handler_base.hpp]: Where is `kSelfTierId` being used? The routing tier shouldn't have a tier id.

**Review by vsreekanti — CHANGES_REQUESTED** [review]
Seems like 3 of the 5 test files are empty. Is there anything that's blocking writing those tests?

**Comment by saurav-c** [comment]
Unit tests have been added for all routing handlers.

**Review by vsreekanti — APPROVED** [review]

**Comment by vsreekanti** [comment]
Awesome, great job!

**Pull request merged**